akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala

```diff
@@ -224,6 +225,73 @@ class ClusterMessageSerializer(val system: ExtendedActorSystem) extends BaseSeri
     }
   }
 
+  private def deserializeJoin(bytes: Array[Byte]): InternalClusterAction.Join = {
+    val m = cm.Join.parseFrom(bytes)
+    val roles = Set.empty[String] ++ m.getRolesList.asScala
+    InternalClusterAction.Join(
+      uniqueAddressFromProto(m.getNode),
+      if (roles.exists(_.startsWith(ClusterSettings.DcRolePrefix))) roles
+      else roles + (ClusterSettings.DcRolePrefix + ClusterSettings.DefaultDataCenter)
+    )
+  }
+
+  private def deserializeWelcome(bytes: Array[Byte]): InternalClusterAction.Welcome = {
+    val m = cm.Welcome.parseFrom(decompress(bytes))
+    InternalClusterAction.Welcome(uniqueAddressFromProto(m.getFrom), gossipFromProto(m.getGossip))
+  }
+
+  private def deserializeLeave(bytes: Array[Byte]): ClusterUserAction.Leave = {
+    ClusterUserAction.Leave(addressFromBinary(bytes))
+  }
+
+  private def deserializeDown(bytes: Array[Byte]): ClusterUserAction.Down = {
+    ClusterUserAction.Down(addressFromBinary(bytes))
+  }
+
+  private def deserializeInitJoin(bytes: Array[Byte]): InternalClusterAction.InitJoin = {
+    val m = cm.InitJoin.parseFrom(bytes)
+    if (m.hasCurrentConfig)
+      InternalClusterAction.InitJoin(ConfigFactory.parseString(m.getCurrentConfig))
+    else
+      InternalClusterAction.InitJoin(ConfigFactory.empty)
+  }
+
+  private def deserializeInitJoinAck(bytes: Array[Byte]): InternalClusterAction.InitJoinAck = {
+    try {
+      val i = cm.InitJoinAck.parseFrom(bytes)
+      val configCheck =
+        if (i.hasConfigCheck) {
+          i.getConfigCheck.getType match {
+            case cm.ConfigCheck.Type.CompatibleConfig   ⇒ CompatibleConfig(ConfigFactory.parseString(i.getConfigCheck.getClusterConfig))
+            case cm.ConfigCheck.Type.IncompatibleConfig ⇒ IncompatibleConfig
+            case cm.ConfigCheck.Type.UncheckedConfig    ⇒ UncheckedConfig
+          }
+        } else UncheckedConfig
+
+      InternalClusterAction.InitJoinAck(addressFromProto(i.getAddress), configCheck)
+    } catch {
+      case ex: akka.protobuf.InvalidProtocolBufferException ⇒
+        // nodes previous to 2.5.9 sends just an address
```

Comment:
Were it not added as optional field?

Reply:
Somewhere there is supposed to be a special case for 2.5.10 joining 2.5.9 that ignores the config check. I don’t know how that was implemented and if the range must be extended to 2.5.11 joining 2.5.9. I think it was that the config was empty (not included), but that wouldn’t have worked if exc was thrown. perhaps ask Renato. (Sorry, on phone).

Also what will happen now when joining 2.5.11 to 2.5.10 (in case someone started with 2.5.10)?

Reply:
The problem is that 2.5.9 sends just an `Address` protobuf message back with the class/string manifest saying `InitJoinAck`, the `InitJoinAck` protobuf message was added in the config check PR, the optional field does not help, I guess it could even be required.

With 2.5.10 joining 2.5.11 both will be sending the `InitJoinAck` protobuf message and all should be working just fine.

Reply:
That might be a reminder to use protobuf wrappers in more cases so it will be easier to extend later on? On the other hand, I think I also argued the other way round not to wrap stuff if not needed to safe space and use manifests for evolving the protocol.

Reply:
Yeap, not sure there was anything wrong with the send-only-address, the issue was more using the old class-based serializer IMO. 

I wish there was a better API for "is this bytes this kind of protobuf message?" than catching an exception though.

Reply:
I think the intention was that the address bytes would be possible to parse as a InitJoinAck proto since it also has an address field first.

But not that 2.5.10 may send another manifest string that is not understood in 2.5.11. That was maybe solved by the manifest2 that I said should be removed, and then it must be added back again with a valid comment.

Reply:
I think we are mixing up classes here. 

`InitJoin` was a singleton in 2.5.9 and became a class in 2.5.10 so for that the manifest changed. 

`InitJoinAck` was always a class so the manifest never changed, it was serialized as adress only in 2.5.9 though, which the cluster config check PR did not handle correctly - this was another part of the regression not reported in the issue but found when I was manually joining a 2.5.9 cluster from this PR.

Reply:
Ok, i might have commented on wrong lines but i’m pretty sure 2.5.10 will not be able to join 2.5.11 because it will send InitJoin manifest that is not understood. Perhaps not important to support that direction.

Reply:
Alright, now I got it, that is correct, rolling downgrade will not work.